### PR TITLE
Preserve `pagination-hotkey` after navigation

### LINKED ISSUE #6146: `pagination-hotkey` is lost after the first navigation on notifications
### Description

The notifications page is dynamic so [any feature that is enabled on it ](https://cs.github.com/refined-github/refined-github?q=pageDetect.isNotifications)should use the observer.

Related code:

https://github.com/refined-github/refined-github/blob/4939c6a2862f9ddcb138874efc627e9b1b8d97ff/source/features/pagination-hotkey.tsx#L24-L25

### How to replicate the issue + URL

1. Open https://github.com/notifications
2. Select 10+ items
3. Mark as done
4. Wait for the DOM to be automatically regenerated (and our changes lost)

### Extension version

22.11.6

### Browser(s) used

Chrome, Safari

### PR DESCRIPTION
**Pull request by yakov116**
Fixes #6146 

## Test URLs
How to replicate the issue + URL
Open https://github.com/notifications
Select 10+ items
Mark as done
Wait for the DOM to be automatically regenerated (and our changes lost)

## Screenshot
NA

### CODE REVIEW

File: source/features/pagination-hotkey.tsx [hunk 1]
```diff
@@ -20,9 +20,13 @@ const previousPageButtonSelectors = [
 	'.prh-commit > .BtnGroup > :first-child', // PR Commits
 ];
 
-function init(): void {
-	addHotkey(select(nextPageButtonSelectors), 'ArrowRight');
-	addHotkey(select(previousPageButtonSelectors), 'ArrowLeft');
+function init(signal: AbortSignal): void {
+	observe(nextPageButtonSelectors, button => {
+		addHotkey(button as HTMLButtonElement | HTMLAnchorElement, 'ArrowRight');
```
Comment (fregante) [inline source/features/pagination-hotkey.tsx:25]: I think you can add the actual tag name to the arrays _and_ append `as const`. e.g.

```ts
const previousPageButtonSelectors = [
	'a.previous_page', // Issue/PR list, Search
	'.paginate-container.BtnGroup > a:first-child', // Notifications
	'.paginate-container > .BtnGroup > a:first-child', // Commits
	'.paginate-container > .pagination > a:first-child', // Releases
 	'.prh-commit > .BtnGroup > a:first-child', // PR Commits
 ] as const
```

Double check the tag names on each page though

Comment (yakov116) [reply source/features/pagination-hotkey.tsx]: I did that Typescript still fails

**Comment by yakov116** [comment]
🤷 not sure where I went wrong here.

**Comment by fregante** [comment]
1. You did not even copy-paste my suggestion
2. The selectors are wrong, there's no `a.paginate-container`
3. You did not run it before commenting "it doesn't work", meaning you're asking me to debug your PRs before you tried

**Comment by yakov116** [comment]
That was an honest mistake I tested each every one, when I was putting in the `a`each selector I by mistake put the a in the wrong place.

Your right I have to be more careful in the future

**Pull request closed without merging**